Tidy JunitXmlReporter

aSemy opened the pull request:
- Update KDoc.
- Pass in a Clock, so the time can be customised.
- Minor tidying to make the code more Kotlin-esque (although I'll concede that's a personal opinion)

I made these refactorings while looking into #4298, and while they didn't turn out to be necessary (I wanted to pass in a clock so that I could run the JUnit XML report tests programatically), I think the refactorings are still useful.

AlexCue987 reviewed (commented):
is it still a draft?

aSemy committed:
Tidy JunitXmlReporter

- Update KDoc.
- Pass in a Clock, so the time can be customised.
- Compute `hostname` with `lazy {}`, to avoid re-evalutating it.
- Minor tidying to make the code more Kotlin-esque (although I'll concede that's a personal opinion)
